### PR TITLE
Implement AnchorBehavior for the Javascript Platform

### PR DESCRIPTION
**Pull request by 4ian**
This is a port of the AnchorBehavior for the Javascript platform.

@victorlevasseur You can take a look if you wish :)

### CODE REVIEW

**Comment by 4ian** [comment]
@victorlevasseur I think the behavior would be even more useful if the anchor were relative to the original size of the game window. For example, in the editor you put a few objects with anchor so that they stick to the left/right/top/bottom of the window.
Right now, if you change the size of the window (for example to adapt to the screen size, especially on mobile phones), objects with anchor won't have a proper position.

What I'm going to do is add an option "Relative to window original size" (maybe checked by default). The behavior will then, if the option is activated, use the original size of the window to compute the distance to the edges.
So at the end, if you put an object in the editor at 10 pixels from the right edge of the window, and add an anchor behavior with the left anchor set to the window right edge, the object will be put at 10 pixels from the right edge of the window if the window still have the original size AND also if the window was resized.

I hope all of this make sense :D 

Basically what I want is to put objects on the scene editor and make sure they stick to edges of the game window even if the window size changed. By having an option, that is checked by default and that will use the default window size for doing the distance computation (when `m_invalidDistances` is `true`), I'll have the expected result :)

**Comment by victorlevasseur** [comment]
I don't understand. Isn't it what the behavior is already doing ?

**Comment by victorlevasseur** [comment]
Ok, understood. You want to keep track of the original window size and use it to set the position of objects created during the game, isn't it ?

**Comment by 4ian** [comment]
Let's take an example :

* Window size in the editor is 400 pixel width.
* The object X position is 350.
* The behavior is set so that the left edge anchor is `window right`.

When the scene starts, `m_invalidDistances` is `true` so `m_leftEdgeDistance` is computed and set to `400 - 350 = 50`. Afterwards,`m_invalidDistances` is `false`, so the object X position is set to `static_cast<float>(windowSize.x) - m_leftEdgeDistance`, which equals `400 - 50 = 350`.
So the object stays at 350 px on the X axis. So far so good.

Now, imagine I resize the window (with an action) so that the width of the scene is 600, and start again the scene. 
`m_invalidDistances` is `true` so `m_leftEdgeDistance` is computed and set to `600 - 350 = 250`. Afterwards, `m_invalidDistances` is `false`, so the object X position is set to `static_cast<float>(windowSize.x) - m_leftEdgeDistance`, which equals `600 - 250 = 350`.
So the object stays at 350 px on the X axis which is **not** what I want => I wanted the object to be at 50 pixels from the right side.

I can "fix" this by always computed distance, when `m_invalidDistances` is `true`, using the window original size. (so that in the second case, `m_leftEdgeDistance` is still **50**).

**Comment by 4ian** [comment]
> You want to keep track of the original window size and use it to set the position of objects created during the game, isn't it ?

Even objects that are already on the scene at startup, because the size of the window may have been changed and is not the size that was in the editor.

**Comment by 4ian** [comment]
I've implemented the option. Will merge that soon if there is no issue while using it :)
